### DISCUSSION TITLE
Custom plot queries?

### DISCUSSION THREAD

**Discussion by sairus7** [epezent/implot#197] [Q&A] [answered]
How can I change query behaviour?

I want a query that is:
- independent from Ctrl/Shift keys, 
- always visible, 
- always in a VerticalMod, 
- can be dragged only in X-axis direction
- its width can be changed by dragging on quey edges. 
- any plot click outside of the query should place it in the center of clicked X-position.

Should I write a custom element, or modify existing query?

**Comment by sairus7** [comment]
I think this is doable if there were `SetPlotQuery` method in addition to `GetPlotQuery` - then we can check custom conditions and modify query accordingly. Didn't find such method for now.

**Comment by epezent** [comment, marked as answer]
@sairus7 , I have added a `SetPlotQuery` for you. Please see the latest master. I'm not sure if this alone will enable the behavior you want, so let me know how it goes. For reference, there are a few other Issues looking for a way to hook into or intervene with ImPlot's input handling (https://github.com/epezent/implot/issues/192, https://github.com/epezent/implot/issues/165). I'm hoping we can do this in a way that facilitates custom plot tools like yours. This will hopefully be worked on soon.
